x-pack/plugins/observability/server/lib/rules/threshold/register_threshold_rule_type.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { ActionGroupIdsOf } from '@kbn/alerting-plugin/common';
+import { IRuleTypeAlerts, RuleType } from '@kbn/alerting-plugin/server';
+import { IBasePath, Logger } from '@kbn/core/server';
+import { legacyExperimentalFieldMap } from '@kbn/alerts-as-data-utils';
+import { createLifecycleExecutor } from '@kbn/rule-registry-plugin/server';
+import { LicenseType } from '@kbn/licensing-plugin/server';
+
+import { thresholdFeatureId } from '../../../../common';
+import { Comparator } from '../../../../common/threshold_rule/types';
+import { OBSERVABILITY_THRESHOLD_RULE_TYPE_ID } from '../../../../common/constants';
+
+import {
+  alertDetailUrlActionVariableDescription,
+  alertStateActionVariableDescription,
+  cloudActionVariableDescription,
+  containerActionVariableDescription,
+  groupByKeysActionVariableDescription,
+  hostActionVariableDescription,
+  labelsActionVariableDescription,
+  metricActionVariableDescription,
+  orchestratorActionVariableDescription,
+  originalAlertStateActionVariableDescription,
+  originalAlertStateWasActionVariableDescription,
+  reasonActionVariableDescription,
+  tagsActionVariableDescription,
+  thresholdActionVariableDescription,
+  timestampActionVariableDescription,
+  valueActionVariableDescription,
+  viewInAppUrlActionVariableDescription,
+} from './messages';
+import {
+  getAlertDetailsPageEnabledForApp,
+  oneOfLiterals,
+  validateIsStringElasticsearchJSONFilter,
+} from './utils';
+import {
+  createMetricThresholdExecutor,
+  FIRED_ACTIONS,
+  WARNING_ACTIONS,
+  NO_DATA_ACTIONS,
+} from './threshold_executor';
+import { ObservabilityConfig } from '../../..';
+import { METRIC_EXPLORER_AGGREGATIONS } from '../../../../common/threshold_rule/constants';
+
+export const MetricsRulesTypeAlertDefinition: IRuleTypeAlerts = {
+  context: OBSERVABILITY_THRESHOLD_RULE_TYPE_ID,
+  mappings: { fieldMap: legacyExperimentalFieldMap },
+  useEcs: true,
```

Comment:
This will add a reference to the ECS component template & legacy alert component template when both are set to true. The legacy alert component template is only needed for rule types that are already on the rule registry prior to 8.8 so I believe that can be set to `false` here. Are you sure you a reference to the full ECS component template for this rule type?

Reply:
@ymao1 updated as discussed.